subiquity/server/controllers/zdev.py

```diff
@@ -703,12 +708,25 @@ async def chzdev_POST(self, action: str, zdev: ZdevInfo) -> List[ZdevInfo]:
 
     async def GET(self) -> List[ZdevInfo]:
         if self.opts.dry_run:
-            return self.zdevinfos.values()
+            return self.dr_zdevinfos.values()
         else:
-            return self.lszdev()
+            return await self.lszdev()
+
+    def _raw_lszdev_sync(self) -> str:
+        return run_command(lszdev_cmd, universal_newlines=True).stdout
 
-    def lszdev(self):
-        devices = run_command(lszdev_cmd, universal_newlines=True).stdout
-        devices = devices.splitlines()
+    async def _raw_lszdev(self) -> str:
+        return (await arun_command(lszdev_cmd)).stdout
+
+    def _parse_lszdev(self, output: str) -> list[ZdevInfo]:
+        devices = output.splitlines()
         devices.sort()
         return [ZdevInfo.from_row(row) for row in devices]
+
+    def lszdev_sync(self) -> list[ZdevInfo]:
+        """Synchronous version of lszdev - which we can drop once we move the
+        call to lszdev outside the initializer."""
+        return self._parse_lszdev(self._raw_lszdev_sync())
```

Comment:
thought: We could potentially refactor how we do the assignment - something along the lines of scheduling another task in `start` - to avoid this sync vs async business? Especially if we only like having the sync version around to support the dry-run scenario. Which is probably what you're getting at with the comment.

Then again, that seems like even more mechanics just to support dry-run and this seems like a perfectly reasonable approach to me. Does this the sync copy have value outside of dry-run? We could just rename to make it clear this is a synchronous helper function for dry-run.

Reply:
Oh yes absolutely a follow-up PR thing, sorry for not making that clearer!